File: Coding Interviews/02 - Basic Knowledge/E365.Count1inBinary.py
class Solution:
    """
    @param: num: An integer
    @return: An integer
    """
    def countOnes(self, num):
        # write your code here
        count = 0
        for i in range(32):
            if num == 0:
                break

            count += num & 1
            num >>= 1

        return count

File: Coding Interviews/02 - Basic Knowledge/E365.Count1inBinary.java
public class Solution {
    /*
     * @param num: An integer
     * @return: An integer
     */
    public int countOnes(int num) {
        // write your code here
        int count = 0;
        while (num != 0) {
            num &= num - 1;
            count++;
        }

        return count;
    }
};
